packages/datepicker-react/src/DatePicker.tsx

```diff
@@ -156,16 +158,26 @@ export function DatePicker({
                     <Label variant={variant} forceCompact={forceCompact}>
                         {label}
                     </Label>
-                    <input
-                        placeholder={placeholder}
-                        type="text"
-                        aria-invalid={!!errorLabel}
-                        className={`jkl-text-field__input`}
-                        data-testid="jkl-datepicker__input"
-                        value={dateString}
-                        onChange={onInputChange}
-                        onFocus={openDatepicker}
-                    />
+                    <div className="jkl-text-field__input-wrapper  jkl-datepicker__input">
+                        <input
+                            placeholder={placeholder}
+                            type="text"
+                            aria-invalid={!!errorLabel}
+                            className="jkl-text-field__input"
+                            data-testid="jkl-datepicker__input"
+                            value={dateString}
+                            onChange={onInputChange}
+                        />
+                        <button
```

Comment:
Since this button toggles the visibility of the calendar it might be good to use the CoreToggle for improved experience for screen readers. 

Have a look at the examples for CoreDatepicker in NRKs documentation: https://static.nrk.no/core-components/latest/index.html?core-datepicker/readme.md

Reply:
Her ønsket jeg å gjenbruke mønstret fra `ActionInputField`, og `core-toggle` passet ikke helt inn der. Tipper det blir lettere med refaktoreringen jeg nevner over, så kan se på det sammen med det 😊

Reply:
Funka fint med `core-toggle`! Eneste bakdelen er at kalenderen ikke lukker seg dersom man tabber seg ut av den (f.eks. til neste inputfelt). Jeg fikk uendelige løkker uansett hvordan jeg prøve å løse det, men det er ikke sikkert det er et så stort problem ettersom man aktivt må velge å åpne kalenderen, og dermed sannsynligvis ender med å faktisk velge en dato eller scape seg ut.